In [2]:
import pandas as pd
import os 

In [3]:
folder_results = "msn/downstream/"

# list all folders in folder_results
folders = os.listdir(folder_results)

In [7]:
# read all the "xgboost_results.csv" from the folders

dfs = {}
for folder in folders:
    file_path = os.path.join(folder_results, folder, "xgboost_results.csv")
    if os.path.exists(file_path):
        df = pd.read_csv(file_path)
        dfs[folder] = df

In [9]:
# elementos que queremos separar
elementos = ['B', 'Cu', 'Zn', 'Fe', 'S', 'Mn']

# dicionário para armazenar os 6 DataFrames finais
dfs_elementos = {}

for elem in elementos:
    linhas = []
    for modelo, df in dfs.items():
        # seleciona apenas a linha referente ao elemento
        linha = df[df['Elemento'] == elem].copy()
        linha['Modelo'] = modelo
        linhas.append(linha[['Modelo', 'R2', 'MAE', 'RMSE']])
    
    # concatena e define o índice como o nome do modelo
    df_final = pd.concat(linhas).set_index('Modelo')
    dfs_elementos[elem] = df_final

In [13]:
def bold_best(df):
    df = df.copy()
    
    # melhores critérios
    best_r2 = df["R2"].max()
    best_mae = df["MAE"].min()
    best_rmse = df["RMSE"].min()
    
    # aplica markdown
    df["R2"] = df["R2"].apply(lambda x: f"**{x:.6f}**" if x == best_r2 else f"{x:.6f}")
    df["MAE"] = df["MAE"].apply(lambda x: f"**{x:.6f}**" if x == best_mae else f"{x:.6f}")
    df["RMSE"] = df["RMSE"].apply(lambda x: f"**{x:.6f}**" if x == best_rmse else f"{x:.6f}")
    
    return df

# -------------------------------------------------------------------
# 4. Imprimir todas as tabelas
# -------------------------------------------------------------------
for elem in elementos:
    print(f"\n### Elemento: {elem}\n")
    display(bold_best(dfs_elementos[elem]))


### Elemento: B



R2           MAE          RMSE
Modelo                                                                   
proto10_embed32_bwpe_SPEC_12H        0.544288      0.116924      0.156575
proto10_embed32_pe_SPEC_15H          0.589413      0.110000      0.148621
proto10_embed32_bwpe_BOTH_12H3F      0.586622      0.114388      0.149125
proto10_embed32_pe_BOTH_12H3F    **0.621741**  **0.107269**  **0.142650**


### Elemento: Cu



R2           MAE          RMSE
Modelo                                                                   
proto10_embed32_bwpe_SPEC_12H    **0.332770**      0.277270  **0.371632**
proto10_embed32_pe_SPEC_15H          0.295612  **0.274983**      0.381840
proto10_embed32_bwpe_BOTH_12H3F      0.307302      0.277600      0.378658
proto10_embed32_pe_BOTH_12H3F        0.309893      0.277118      0.377949


### Elemento: Zn



R2           MAE          RMSE
Modelo                                                                   
proto10_embed32_bwpe_SPEC_12H        0.451792      1.019267      1.415902
proto10_embed32_pe_SPEC_15H          0.519743      0.915126      1.325249
proto10_embed32_bwpe_BOTH_12H3F      0.421043      0.963680      1.455070
proto10_embed32_pe_BOTH_12H3F    **0.540896**  **0.906421**  **1.295734**


### Elemento: Fe



R2            MAE           RMSE
Modelo                                                                     
proto10_embed32_bwpe_SPEC_12H        0.383498      31.807866      44.029783
proto10_embed32_pe_SPEC_15H          0.499417      29.867836      39.675063
proto10_embed32_bwpe_BOTH_12H3F      0.479150      30.154246      40.470243
proto10_embed32_pe_BOTH_12H3F    **0.555491**  **28.286069**  **37.386936**


### Elemento: S



R2           MAE           RMSE
Modelo                                                                    
proto10_embed32_bwpe_SPEC_12H    **0.416865**  **6.710094**  **10.384232**
proto10_embed32_pe_SPEC_15H          0.394532      6.775778      10.581208
proto10_embed32_bwpe_BOTH_12H3F      0.407482      7.059170      10.467443
proto10_embed32_pe_BOTH_12H3F        0.401525      6.742780      10.519926


### Elemento: Mn



R2            MAE           RMSE
Modelo                                                                     
proto10_embed32_bwpe_SPEC_12H        0.473751      11.799570      16.368791
proto10_embed32_pe_SPEC_15H      **0.488882**  **11.576266**  **16.131740**
proto10_embed32_bwpe_BOTH_12H3F      0.454327      12.177160      16.668142
proto10_embed32_pe_BOTH_12H3F        0.453366      12.047442      16.682814

In [16]:
!pip  install tabulate


[notice] A new release of pip is available: 23.0.1 -> 25.3
[notice] To update, run: pip install --upgrade pip


In [17]:
def bold_best_markdown(df):
    df = df.copy()

    best_r2 = df["R2"].max()
    best_mae = df["MAE"].min()
    best_rmse = df["RMSE"].min()

    def fmt(val, best):
        return f"**{val:.4f}**" if val == best else f"{val:.4f}"

    df["R2"] = df["R2"].apply(lambda x: fmt(x, best_r2))
    df["MAE"] = df["MAE"].apply(lambda x: fmt(x, best_mae))
    df["RMSE"] = df["RMSE"].apply(lambda x: fmt(x, best_rmse))

    return df

# ---------------------------------------------------------
# 4. Imprimir tabelas em Markdown
# ---------------------------------------------------------
for elem in elementos:
    print(f"### Elemento {elem}\n")
    print(bold_best_markdown(dfs_elementos[elem]).to_markdown())
    print("\n")

### Elemento B

| Modelo                          | R2         | MAE        | RMSE       |
|:--------------------------------|:-----------|:-----------|:-----------|
| proto10_embed32_bwpe_SPEC_12H   | 0.5443     | 0.1169     | 0.1566     |
| proto10_embed32_pe_SPEC_15H     | 0.5894     | 0.1100     | 0.1486     |
| proto10_embed32_bwpe_BOTH_12H3F | 0.5866     | 0.1144     | 0.1491     |
| proto10_embed32_pe_BOTH_12H3F   | **0.6217** | **0.1073** | **0.1426** |


### Elemento Cu

| Modelo                          | R2         | MAE        | RMSE       |
|:--------------------------------|:-----------|:-----------|:-----------|
| proto10_embed32_bwpe_SPEC_12H   | **0.3328** | 0.2773     | **0.3716** |
| proto10_embed32_pe_SPEC_15H     | 0.2956     | **0.2750** | 0.3818     |
| proto10_embed32_bwpe_BOTH_12H3F | 0.3073     | 0.2776     | 0.3787     |
| proto10_embed32_pe_BOTH_12H3F   | 0.3099     | 0.2771     | 0.3779     |


### Elemento Zn

| Modelo                          | R2        

### Elemento B

| Modelo                          | R2         | MAE        | RMSE       |
|:--------------------------------|:-----------|:-----------|:-----------|
| proto10_embed32_bwpe_SPEC_12H   | 0.5443     | 0.1169     | 0.1566     |
| proto10_embed32_pe_SPEC_15H     | 0.5894     | 0.1100     | 0.1486     |
| proto10_embed32_bwpe_BOTH_12H3F | 0.5866     | 0.1144     | 0.1491     |
| proto10_embed32_pe_BOTH_12H3F   | **0.6217** | **0.1073** | **0.1426** |


### Elemento Cu

| Modelo                          | R2         | MAE        | RMSE       |
|:--------------------------------|:-----------|:-----------|:-----------|
| proto10_embed32_bwpe_SPEC_12H   | **0.3328** | 0.2773     | **0.3716** |
| proto10_embed32_pe_SPEC_15H     | 0.2956     | **0.2750** | 0.3818     |
| proto10_embed32_bwpe_BOTH_12H3F | 0.3073     | 0.2776     | 0.3787     |
| proto10_embed32_pe_BOTH_12H3F   | 0.3099     | 0.2771     | 0.3779     |


### Elemento Zn

| Modelo                          | R2         | MAE        | RMSE       |
|:--------------------------------|:-----------|:-----------|:-----------|
| proto10_embed32_bwpe_SPEC_12H   | 0.4518     | 1.0193     | 1.4159     |
| proto10_embed32_pe_SPEC_15H     | 0.5197     | 0.9151     | 1.3252     |
| proto10_embed32_bwpe_BOTH_12H3F | 0.4210     | 0.9637     | 1.4551     |
| proto10_embed32_pe_BOTH_12H3F   | **0.5409** | **0.9064** | **1.2957** |


### Elemento Fe

| Modelo                          | R2         | MAE         | RMSE        |
|:--------------------------------|:-----------|:------------|:------------|
| proto10_embed32_bwpe_SPEC_12H   | 0.3835     | 31.8079     | 44.0298     |
| proto10_embed32_pe_SPEC_15H     | 0.4994     | 29.8678     | 39.6751     |
| proto10_embed32_bwpe_BOTH_12H3F | 0.4791     | 30.1542     | 40.4702     |
| proto10_embed32_pe_BOTH_12H3F   | **0.5555** | **28.2861** | **37.3869** |


### Elemento S

| Modelo                          | R2         | MAE        | RMSE        |
|:--------------------------------|:-----------|:-----------|:------------|
| proto10_embed32_bwpe_SPEC_12H   | **0.4169** | **6.7101** | **10.3842** |
...
| proto10_embed32_bwpe_BOTH_12H3F | 0.4543     | 12.1772     | 16.6681     |
| proto10_embed32_pe_BOTH_12H3F   | 0.4534     | 12.0474     | 16.6828     |


Output is truncated. View as a scrollable element or open in a text editor. Adjust cell output settings...